llama_index/indices/response/response_builder.py

```diff
@@ -622,5 +689,12 @@ def get_response_builder(
             simple_template=simple_template,
             streaming=streaming,
         )
+    elif mode == ResponseMode.ACCUMULATE:
+        return Accumulate(
+            service_context=service_context,
+            text_qa_template=text_qa_template,
+            streaming=streaming,
+            separator=separator,
```

Comment:
Hmm, my only issue with this is that we will also need to update wherever `get_response_builder` is called to include this extra kwarg. Right now, the user can't actually set this from calling `as_query_engine()` 🤔

Looking at the code, maybe it would be better to put this in `get_response` so that the user kwargs can easily make their way to it? See this section of code, we could leverage response_kwargs here: https://github.com/jerryjliu/llama_index/blob/main/llama_index/indices/query/response_synthesis.py#L163

Maybe you'll notice another way to do it too (or maybe it is actually set-able by the user and I missed that haha) :)

Reply:
🤯 
That was as great suggestion, thank you! I'm just in my first week of using llama_index so I'm still learning how data is being passed down through various classes and was unaware of the power of response_kwargs. Check it out now and let me know what you think. Definitely a big improvement over having to pass everything through `get_response_builder` imho.

Reply:
Amazing, much smoother I agree 💪lots of kwargs get passed everywhere so it's super easy to miss haha 

With this latest change, now the user can set the mode just by doing `index.as_query_engine(response_mode="accumulate", separator="my_sep\n")`